### PR TITLE
Is unique custom condition

### PR DESCRIPTION
**Pull request by half2me**
Thought this might be useful. A small addition to the `isUnique` rule, that allows it to be optionally extended to use custom conditions.

### CODE REVIEW

File: src/ORM/Rule/IsUnique.php [hunk 1]
```diff
@@ -79,6 +79,10 @@ public function __invoke(EntityInterface $entity, array $options)
             }
         }
 
+        if (array_key_exists('conditions', $options)) {
```
Comment (dereuromark) [inline src/ORM/Rule/IsUnique.php]: why not just isset()?

**Comment by dereuromark** [comment]
The "### Options" part also needs to be updated for documentation.
But otherwise looks good to me.

**Comment by half2me** [comment]
> ### Options
Its just a bit strange though that the `$options` is being used in two different places: one in the constructor, and another time in `__invoke()`. For example the existing option for `allowMultipleNulls` seems like it can only be set in the constructor, even though it might be expected to work on invocation as well. Where should I put my option?

**Comment by dereuromark** [comment]
Do we want to continue with this for 3.x?

**Comment by dereuromark** [comment]
Bump - this does look useful if you want to finish it up.

**Comment by saeideng** [comment]
@dereuromark  could this be part of `$rules->add()` and not rule itself (`isUnique`,.....)?

**Comment by othercorey** [comment]
Closing due to no activity. For anyone interested, please open an issue to discuss.

**Pull request closed without merging**